src/libraries/StringParser.sol

```diff
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import {VmSafe} from "forge-std/Vm.sol";
+import {LibString} from "@solady/utils/LibString.sol";
+
+/// @title StringParser
+/// @notice Library for parsing string representations of decimal numbers
+library StringParser {
+    using LibString for string;
+
+    /// @notice Struct to hold parsed amount components
+    struct AmountComponents {
+        uint256 wholePart;
+        uint256 decimalPart;
+        uint8 decimalPlaces;
+        bool hasDecimals;
+    }
+
+    /// @notice Validates that the amount string has a valid format
+    /// @param components The split components of the amount string
+    /// reverts if components are invalid
+    function validateAmountFormat(string[] memory components) internal pure {
+        // Check for invalid format (more than one decimal point)
+        require(components.length <= 2, "invalid amount");
```

Comment:
Can you do the same with all the require functions in this file (comment above).